feat(msal): Create a new add-in for MSAL (#3342)
unoplatform/uno: doc/articles/interop/MSAL.md

## General usage

To use MSAL into a Uno project, follow the following steps:

1. Add a reference to [`Uno.UI.MSAL`](https://www.nuget.org/packages/Uno.UI.MSAL) package to all your heads - including UWP.

2. Follow [Microsoft Documentation](https://docs.microsoft.com/azure/active-directory/develop/msal-net-initializing-client-applications) to integrate with your app.

3. Change the `IPublicCLientApplication` initialization to add a call to `.WithUnoHelpers()` like this:

   ``` csharp
   IPublicClientApplication _app = PublicClientApplicationBuilder.Create(clientId)
       [...]
       .WithUnoHelpers() // Add this line before the .Build()
       .Build();
   ```

4. Where you are using the _Interactive_ mode (`_app.AcquireTokenInteractive`), add another call to `.WithUnoHelpers()` like this:

   ``` csharp
   var authResult = await _app.AcquireTokenInteractive(scopes)
       .WithPrompt(Prompt.SelectAccount)
       [...]
       .WithUnoHelpers() // Add this line on interactive token acquisition flow
       .ExecuteAsync();
   ```

By adding those helpers, Uno will correctly add required initializations to MSAL for all supported platforms.

## Windows - UWP

> Nothing to do on UWP. The `.WithUnoHelpers()` does nothing on UAP/UWP platforms, they are just there to allow the code to compile without introducing ugly `#if` in your code.

Reviewer: ```suggestion
There is nothing to change for UWP. The `.WithUnoHelpers()` does nothing on UAP/UWP platforms, they are just there to allow the code to compile without introducing `#if` conditionals in your code.
```

## Android

> You'll need to setup the Return URI following the Microsoft documentation:

Reviewer: ```suggestion
You'll need to setup the return URI following the Microsoft documentation:
```

* Official documentation <https://docs.microsoft.com/azure/active-directory/develop/msal-net-xamarin-android-considerations>

* Wiki https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Xamarin-Android-specifics

> * **No need** to call `.WithParentActivity()`: this is properly initialized by `.WithUnoHelpers()`.

Reviewer: ```suggestion
* There is no need to call `.WithParentActivity()` as it is already initialized by `.WithUnoHelpers()`.
```

## iOS & macOS

Follow Microsoft's documentation: <https://docs.microsoft.com/azure/active-directory/develop/msal-net-xamarin-ios-considerations>.

> * **No need** to call `.WithParentActivity()`: this is properly initialized by `.WithUnoHelpers()`.

Reviewer: ```suggestion
* There is no need to call `.WithParentActivity()` as it is already initialized by `.WithUnoHelpers()`.
```

- Token cache is _in-memory_ for now­. The library is not persisting the token anywhere in the browser yet. The app can save it.

> Important: when making http requests from Wasm, don't forget to use the `WasmHttpHandler` on the `HttpClient`, or you'll get this error: `Operation is not supported on this platform.`. [More details here to fix that error](https://platform.uno/docs/articles/faq.html#is-it-possible-to-make-http-web-requests-using-the-wasm-target).

Reviewer: ```suggestion
Important: when making HTTP requests from Wasm, don't forget to use the `WasmHttpHandler` on the `HttpClient`, or you'll get this error: `Operation is not supported on this platform.`. [More details here to fix that error](https://platform.uno/docs/articles/faq.html#is-it-possible-to-make-http-web-requests-using-the-wasm-target).
```
